### DISCUSSION TITLE
I get an errror, from time to time

### DISCUSSION THREAD

**Discussion by amelcon-dot** [abraunegg/onedrive#1391] [Q&A] [answered]
Hello,
Thanks a lot for continuing the work on this sync client. I managed to setup the client, and I must admit that it is working pretty well.
However, this morning I got this error in a number of files:
```
Uploading modified file v1/MEGA/upct/bin/V1_amelcon_Local_Mail.txt ...
skipped.
ERROR: Microsoft OneDrive API returned an error with the following message:
  Error Message:    HTTP request returned status code 412 (Precondition Failed)
  Error Reason:     ETag does not match current item's value
  Error Timestamp:  2021-04-07T10:49:04
  API Request ID:   d8a52774-cf4c-43d2-868c-52c83a63bb11
  Calling Function: uploadFileDifferences()
```
The only way to stop the error is to add --resync. However, this is very slow.
Do you know what triggers the error, and what can I do to prevent from happening in the future?.
Thanks a lot.

**Comment by amelcon-dot** [comment]
Just in case it is useful, here I attach my configuration:
```
Configuration file successfully loaded
onedrive version                       = v2.4.10
Config path                            = /home/amelcon/.config/onedriveAMELCON
Config file found in config path       = true
Config option 'check_nosync'           = true
Config option 'sync_dir'               = /home/amelcon
Config option 'skip_dir'               = *.aedtresults|*_lib
Config option 'skip_file'              = ~*|.~*|*.tmp|*.aux|*.bbl|*.blg|*.dvi|*.log|*.soc|Thumbs.db*
Config option 'skip_dotfiles'          = false
Config option 'skip_symlinks'          = true
Config option 'monitor_interval'       = 300
Config option 'min_notify_changes'     = 5
Config option 'log_dir'                = /home/amelcon/logs/onedriveAMELCON/
Config option 'classify_as_big_delete' = 1000
Config option 'upload_only'            = false
Config option 'no_remote_delete'       = false
Config option 'remove_source_files'    = false
Config option 'sync_root_files'        = false
Selective sync 'sync_list' configured  = false
Business Shared Folders configured     = false
```

**Comment by abraunegg** [comment, marked as answer]
@amelcon-dot 
> Do you know what triggers the error, and what can I do to prevent from happening in the future?.

This error is triggered by Microsoft modifying your file post upload. There is nothing that can be done, by you to prevent this, however, the actual 412 response should be correctly handled, potentially an additional 412 check is missing - which is odd ..

Please open a new bug report following the correct support process: https://github.com/abraunegg/onedrive/blob/master/README.md#reporting-an-issue-or-bug

**Reply by amelcon-dot** [reply]
Thanks for the description. I opened a new bug report for the additional 412 check.
After your description, I wonder if this may be related with using rclone to upload certain files, and then conflicting with onedrive on the next --synchronize operation. In this case it may be not good idea to use both tools on the same folder?.
Thanks again.

**Reply by abraunegg** [reply]
> After your description, I wonder if this may be related with using rclone to upload certain files, and then conflicting with onedrive on the next --synchronize operation. In this case it may be not good idea to use both tools on the same folder?

Yes - this is most certainly potentially related to using 'rclone' and this client on the same directory - but still the 412 error regardless of client use is not being handled correctly. The PR I have provided *should* resolve this. My advice is use one client, not both. I cannot tell you how rclone works / handles conflicts / resolves conflicts ... but I suggest you only use one client, not two.

**Reply by amelcon-dot** [reply]
Thanks I will do as you suggest. Thanks for the support.
